lib/broadway/producer.ex

```diff
@@ -58,6 +65,17 @@ defmodule Broadway.Producer do
     end
   end
 
+  @impl true
+  def handle_cast(:cancel_producer, state) do
+    %{module: module, module_state: module_state} = state
+
+    if function_exported?(module, :cancel, 1) do
+      module.cancel(module_state)
```

Comment:
I would like to name this something a bit more distinct. For example, we already have `handle_cancel` from GenStage so this can be confused with that.

We could call it drain, but we aren't quite draining. Any other suggestions?

Reply:
How about `suspend`? It sounds closer to what actually happens. We could even have a `resume` in the future, if we need.

Reply:
That's what makes sense from the RabbitMQ perspective but I am not sure from the Broadway perspective. From Broadway, this sounds more like "prepare_for_draining". Maybe `handle_drain`? It is a callback invoked before the draining starts. Ah, we will also need docs for this and declare it as a proper optional callback on Broadway.Producer.

Reply:
`handle_drain` sounds like it would do the draining itself, which it won't, so I would go with something like `prepare_for_draining` or `before_draining`.

Reply:
Let's go with `prepare_for_draining` then!